blockchain/blockchain.go

```diff
@@ -409,12 +409,18 @@ func (bc *blockchain) MintNewBlock(timestamp time.Time) (*block.Block, error) {
 	return &blk, nil
 }
 
-//  CommitBlock validates and appends a block to the chain
+// CommitBlock validates and appends a block to the chain
 func (bc *blockchain) CommitBlock(blk *block.Block) error {
-	bc.mu.Lock()
-	defer bc.mu.Unlock()
 	timer := bc.timerFactory.NewTimer("CommitBlock")
 	defer timer.End()
+
+	if err := bc.ValidateBlock(blk); err != nil {
+		return err
+	}
```

Comment:
`ValidateBlock(blk)` is already called before calling `CommitBlock()` here, it's not as simple as this.
You can chat with haixiang to understand more about what the problem is, and what exactly needs to be done